src/js/common/schemaform/validation.js

```diff
@@ -343,31 +343,39 @@ export function validateDateRange(errors, dateRange, formData, schema, errorMess
   }
 }
 
-export function validateFileField(errors, fileList) {
-  fileList.forEach((file, index) => {
-    let error;
-    if (file.errorMessage) {
-      error = file.errorMessage;
-    } else if (file.uploading) {
-      error = 'Uploading file...';
-    } else if (!file.confirmationCode) {
-      error = 'Something went wrong...';
-    }
+function setError(file, index, errors) {
+  let error;
+  if (file.errorMessage) {
+    error = file.errorMessage;
+  } else if (file.uploading) {
+    error = 'Uploading file...';
+  } else if (!file.confirmationCode) {
+    error = 'Something went wrong...';
+  }
 
-    if (error && !errors[index]) {
-      /* eslint-disable no-param-reassign */
-      errors[index] = {
-        __errors: [],
-        addError(msg) {
-          this.__errors.push(msg);
-        }
-      };
-      /* eslint-enable no-param-reassign */
-    }
+  if (error && !errors[index]) {
+    /* eslint-disable no-param-reassign */
+    errors[index] = {
+      __errors: [],
+      addError(msg) {
+        this.__errors.push(msg);
+      }
+    };
+    /* eslint-enable no-param-reassign */
+  }
 
-    if (error) {
-      errors[index].addError(error);
-    }
+  if (error) {
+    errors[index].addError(error);
+  }
+}
+
+export function validateFile(errors, file) {
+  setError(file, 0, errors);
```

Comment:
I don't think this will work, there's no need for an index if the field is a regular object. You would want to call `errors.addError` for this field to set the message correctly.

I was thinking that this could be the reusable function:

```js
export function getFileError(file) {
  if (file.errorMessage) {
    return file.errorMessage;
  } else if (file.uploading) {
    return 'Uploading file...';
  } else if (!file.confirmationCode) {
    return 'Something went wrong...';
  }

  return null;
}
```

Then you can leave validateFileField the way it was before, except for using that function to replace the above logic.

And for the validation for a single file field, I think that's specific to the photo field right now, so I would take it out of this file and put it in a validation.js file in the vic-v2 folder. You can have it call the getErrorMessage function that would be in here. The square image validation would go in that new file, too, but that can be left to the other a11y ticket, I think.

Reply:
Thanks @jbalboni that's a great idea (updated).